Sources/ConcordiumSwiftSDK/Model/Account.swift

```diff
@@ -16,9 +49,373 @@ struct AccountAddress {
+                bakerAggregationVerifyKey: grpc.aggregationKey.value
+        )
+    }
+}
+
+enum StakePendingChange {
+    case reduceStake(newStake: Amount, effectiveTime: Date)
+    case removeStake(effectiveTime: Date)
+
+    static func fromGrpcType(_ grpc: Concordium_V2_StakePendingChange) -> StakePendingChange? {
+        switch grpc.change {
+        case nil:
+            return nil
+        case let .reduce(r):
+            return .reduceStake(newStake: r.newStake.value, effectiveTime: dateFromUnixTimeMillis(r.effectiveTime.value))
+        case let .remove(r):
+            return .removeStake(effectiveTime: dateFromUnixTimeMillis(r.value))
+        }
+    }
+}
+
+struct AmountFraction {
+    let partsPerHundredThousand: UInt32
+
+    static func fromGrpcType(_ grpc: Concordium_V2_AmountFraction) -> AmountFraction {
+        AmountFraction(partsPerHundredThousand: grpc.partsPerHundredThousand)
+    }
+}
+
+enum OpenStatus: Int {
+    case openForAll = 0
+    case closedForNew = 1
+    case closedForAll = 2
+
+    static func fromGrpcType(_ grpc: Concordium_V2_OpenStatus) -> OpenStatus {
+        OpenStatus(rawValue: grpc.rawValue)!
+    }
+}
+
+struct CommissionRates {
+    let finalization: AmountFraction
+    let baking: AmountFraction
+    let transaction: AmountFraction
+
+    static func fromGrpcType(_ grpc: Concordium_V2_CommissionRates) -> CommissionRates {
+        CommissionRates(
+                finalization: .fromGrpcType(grpc.finalization),
+                baking: .fromGrpcType(grpc.baking),
+                transaction: .fromGrpcType(grpc.transaction)
+        )
+    }
+}
+
+struct BakerPoolInfo {
+    let openStatus: OpenStatus
+    let metadataUrl: String
+    let commissionRates: CommissionRates
+
+    static func fromGrpcType(_ grpc: Concordium_V2_BakerPoolInfo) -> BakerPoolInfo {
+        BakerPoolInfo(
+                openStatus: .fromGrpcType(grpc.openStatus),
+                metadataUrl: grpc.url,
+                commissionRates: .fromGrpcType(grpc.commissionRates)
+        )
+    }
+}
+
+enum DelegationTarget {
+    case passive
+    case baker(BakerId)
+
+    static func fromGrpcType(_ grpc: Concordium_V2_DelegationTarget) -> DelegationTarget? {
+        switch grpc.target {
+        case nil:
+            return nil
+        case .passive:
+            return .passive
+        case let .baker(b):
+            return .baker(b.value)
+        }
+    }
+}
+
+enum AccountStakingInfo {
+    case baker(
+            stakedAmount: Amount,
+            restakeEarnings: Bool,
+            bakerInfo: BakerInfo,
+            pendingChange: StakePendingChange?,
+            poolInfo: BakerPoolInfo?
+    )
+    case delegated(
+            stakedAmount: Amount,
+            restakeEarnings: Bool,
+            delegationTarget: DelegationTarget,
+            pendingChange: StakePendingChange?
+    )
+
+    static func fromGrpcType(_ grpc: Concordium_V2_AccountStakingInfo) throws -> AccountStakingInfo? {
```

Comment:
Should we add tests for the different cases?

Reply:
Discussed above.